expander.c

```diff
@@ -240,6 +243,53 @@ bgpq_expander_add_prefix_range(struct bgpq_expander *b, char *prefix)
 	return sx_prefix_range_parse(b->tree, b->family, b->maxlen, prefix);
 }
 
+char*
+bgpq_get_asset(char *object){
+	char *d = strstr(object, "::");
+	if (d){
+		d += 2;
+	} else {
+		d = object;
+	}
+
+	char *ec = strchr(d, ':');
+	if (ec) {
+		ec += 1;
+	} else {
+		ec = d;
+	}
+
+	char *asset = (char*)calloc(1, 256);
```

Comment:
> Check whether the calloc() call succeeded by comparing asset to NULL; if it is NULL terminate the execution with err(1, NULL). In other words, something along these lines:

Ah yeah, my bad!

> No need to cast (char*) unto the calloc() call.

At some point stricter compile flags should be introduced IMO, because there are various issues in this code which are not being picked up. I used cast because if stricter compile flags were to be introduced at some point in the future, this would be one less issue. I have removed the cast now.